osu.Desktop/Updater/SquirrelUpdateManager.cs

```diff
@@ -50,6 +51,10 @@ private void load(INotificationOverlay notifications)
 
         protected override async Task<bool> PerformUpdateCheck() => await checkForUpdateAsync().ConfigureAwait(false);
 
+        // We really don't want to do update stuff during gameplay. Including when the user is in break time / paused.
+        // Checking the toolbar visibility is a nice hacky way of avoiding this for now.
+        private bool isInGameplay => localUserInfo?.IsPlaying.Value == true || game.Toolbar?.State.Value == Visibility.Visible;
```

Comment:
Uh... what? The hackiness of this aside (that I'm not sure I can accept), isn't this backwards? *Hidden* visibility is what you'd want here no to hack-check if we're in gameplay?

The non-hacky way to do this would be to follow something like

https://github.com/ppy/osu/blob/d3f31fad8012ad2531da7c947a46f6aa68bec080/osu.Game/OsuGame.cs#L1148

and check whatever that returns for `Player` specifically.

Reply:
Woudl you prefer the non-hacky route?

The more involved pathway would be adding a second flag to `ILocalUserPlayInfo` rather than create a new component.

Reply:
> Woudl you prefer the non-hacky route?

I would yeah. I don't want to see an issue thread about someone not getting update notifications because they manually toggled the toolbar off.

> The more involved pathway would be adding a second flag to `ILocalUserPlayInfo` rather than create a new component.

But I wouldn't wanna touch this, with how many places use this it'll be a thirty file refactor. We can reconsider if this crops up again.